client/src/components/Format/Format.component.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react'
+
+// ---| core |---
+import { cn } from 'tools'
+
+// ---| pages |---
+// ---| screens |---
+// ---| components |---
+import { Text, TextProps } from 'components/Text'
+
+// ---| self |---
+import css from './Format.module.scss'
+import {
+  DEFAULT_SIZE_OPTIONS, SIZE_OPTIONS, DEFAULT_WEIGHT_OPTIONS, WEIGHT_OPTIONS,
+  toString, toNumber, toUnit, toDate, capitalize, getMaxOptions,
+} from './Format.tool'
+
+export type FormatValue = string | number | Date | undefined | null
+
+const STRING_FORMAT_MAP = {
+  'uppercase': (v: FormatValue) => toString(v).toUpperCase(), // ABC DEF
+  'lowercase': (v: FormatValue) => toString(v), // abc def
+  'capitalize': (v: FormatValue) => {
+    const whitespace = ' '
+
+    return toString(v).split(whitespace).map(capitalize).join(whitespace)
+  }, // Abc Def
+  'title': (v: FormatValue) => capitalize(toString(v)), // Abc def
+}
+
+const NUMBER_FORMAT_MAP = {
+  // 123,456,789.98
+  'number': (v: FormatValue) => toNumber(v, 'decimal', 2),
+
+  // 123,456,789
+  'amount': (v: FormatValue) => toNumber(v, 'decimal', 0),
+
+  // 123,456,789.98 %
+  'percent': (v: FormatValue) => toNumber(v, 'percent', 2),
+
+  // 123.46 M$  - K, M, B
+  'currency': (v: FormatValue) => toNumber(v, 'currency', 2, 'USD', 'compact'),
```

Comment:
it's bad idea to have more then 3 arguments. can we have options as intl-options here?